src/fund_migration.rs

```diff
@@ -19,25 +24,127 @@ pub enum MigrationStatus {
+    sdk: &BreezServices,
+) -> Result<()> {
+    if matches!(
+        data_store
+            .lock()
+            .unwrap()
+            .retrive_funds_migration_status()?,
+        MigrationStatus::Completed | MigrationStatus::NotNeeded
+    ) {
+        return Ok(());
+    }
+
+    let (private_key, public_key) = derive_ldk_keys(seed)?;
+    let public_key = public_key.serialize().to_hex();
+
+    let balance = fetch_balance(public_key.clone())?;
+    if balance == 0 {
+        data_store
+            .lock()
+            .unwrap()
+            .append_funds_migration_status(MigrationStatus::NotNeeded)?;
+        return Ok(());
+    }
+
+    data_store
+        .lock()
+        .unwrap()
+        .append_funds_migration_status(MigrationStatus::Pending)?;
+
+    let lsp_info = sdk.lsp_info().await.map_to_runtime_error(
+        RuntimeErrorCode::LspServiceUnavailable,
+        "Failed to get LSP info",
+    )?;
+    let lsp_fee = lsp_info
+        .opening_fee_params_list
+        .get_cheapest_opening_fee_params()
+        .map_to_permanent_failure("Failed to get LSP fees")?;
+    let amout_to_request = add_lsp_fees(balance, &lsp_fee) * 1_000;
+
+    let invoice = sdk
+        .receive_payment(breez_sdk_core::ReceivePaymentRequest {
+            amount_sats: amout_to_request,
+            description: MIGRATION_DESCRIPTION.to_string(),
+            preimage: None,
+            opening_fee_params: Some(lsp_fee),
+            use_description_hash: None,
+            expiry: None,
+            cltv: None,
+        })
+        .await
+        .map_to_runtime_error(RuntimeErrorCode::NodeUnavailable, "Failed to issue invoice")?;
+    let invoice = invoice.ln_invoice.bolt11;
+    let signature = sign_message(&private_key, &invoice);
+
+    match payout(public_key, invoice, signature) {
+        Ok(()) => data_store
+            .lock()
+            .unwrap()
+            .append_funds_migration_status(MigrationStatus::Completed),
+        Err(e) => {
+            let _ = data_store
+                .lock()
+                .unwrap()
+                .append_funds_migration_status(MigrationStatus::Failed);
+            Err(e)
+        }
+    }
+}
+
+fn payout(_public_key: String, _invoice: String, _signature: String) -> Result<()> {
+    // TODO: Implement.
+    Ok(())
+}
+
+fn fetch_balance(_public_key: String) -> Result<u64> {
+    // TODO: Implement.
+    Ok(0)
+}
+
+fn add_lsp_fees(amount_msat: u64, lsp_fee: &OpeningFeeParams) -> u64 {
+    const MILLION: u64 = 1_000_000;
+
+    const MIN_REQUEST_MSAT: u64 = 1_000;
+    if amount_msat < MIN_REQUEST_MSAT {
+        return lsp_fee.min_msat + MIN_REQUEST_MSAT;
+    }
+
+    let one_minus_proportional = MILLION - lsp_fee.proportional as u64;
+
+    if amount_msat * lsp_fee.proportional as u64 / one_minus_proportional < lsp_fee.min_msat {
+        amount_msat + lsp_fee.min_msat
+    } else {
+        let result = amount_msat * MILLION / one_minus_proportional;
+        result / 1_000 * 1_000
+    }
```

Comment:
Not sure what you're calculating here. I'm coming up with a different calculation which also results in different amounts :thinking: 

Tell me what I'm missing here:

```suggestion
    let proportional_fee = amount_msat * lsp_fee.proportional as u64 / MILLION;

    if proportional_fee < lsp_fee.min_msat {
        amount_msat + lsp_fee.min_msat
    } else {
        let result = amount_msat + proportional_fee;
        result / 1_000 * 1_000
    }
```

Reply:
As discussed. My assumption here is wrong, because this calculation assumes a payment where the _sent_ amount is an exactly defined number, but what we need for the migration is a payment where the _received_ amount is achieved accurately.